File: tests/test_ruby/services/user_service.rb
require_relative '../models/user'
require_relative '../utils/helpers'
require 'logger'

class UserService
  def initialize
    @logger = Logger.new(STDOUT)
  end

  def save(user)
    @logger.info("Saving #{user.name}")
  end

  def find_all
    []
  end
end

File: tests/test_zig/services/user_service.zig
const std = @import("std");
const User = @import("../models/user.zig");

pub const UserService = struct {
    pub fn init() UserService {
        return UserService{};
    }

    pub fn save(self: *UserService, user: *const User.User) void {
        _ = self;
        std.debug.print("Saving user: {s}\n", .{user.name});
    }
};
